config/localTemplate.go

```diff
@@ -744,12 +744,14 @@ func (cfg Local) IsGossipServer() bool {
 
 // IsWsGossipServer returns true if a node configured to run a listening ws net
 func (cfg Local) IsWsGossipServer() bool {
-	return cfg.NetAddress != "" && !cfg.EnableP2P
+	// 1. NetAddress is set and EnableP2P is not set
+	// 2. NetAddress is set and EnableP2PHybridMode is set then EnableP2P is overridden  by EnableP2PHybridMode
+	return cfg.NetAddress != "" && (!cfg.EnableP2P || cfg.EnableP2PHybridMode)
 }
 
 // IsP2PGossipServer returns true if a node configured to run a listening p2p net
 func (cfg Local) IsP2PGossipServer() bool {
-	return cfg.EnableP2P && cfg.NetAddress != "" || cfg.EnableP2PHybridMode && cfg.P2PNetAddress != ""
+	return cfg.EnableP2P && !cfg.EnableP2PHybridMode && cfg.NetAddress != "" || cfg.EnableP2PHybridMode && cfg.P2PNetAddress != ""
```

Comment:
I'd personally put the two sides of the `||` in quotes to help with readability here.
```suggestion
	return (cfg.EnableP2P && !cfg.EnableP2PHybridMode && cfg.NetAddress != "") || (cfg.EnableP2PHybridMode && cfg.P2PNetAddress != "")
```

We should fail to start if hybrid is enabled but the net addresses are not both set right?